test/bpf/check-consts.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+#
+# Copyright 2020 Authors of Cilium
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+SCRIPTDIR="$(dirname $(realpath $0))"
+BPFDIR=$(realpath $SCRIPTDIR/../../bpf/)
+
+
+# object file with the dummy constants so that constchecker can test itself
+XOBJ_SELF=bpf_constchecker_self.o
+# object file with cilium constants
+XOBJ=bpf_constchecker.o
+
+# ensure that objects are up-to-date
+make -C $BPFDIR $XOBJ_SELF $XOBJ
+
+# NB: in both cases, we pass proper flags to go test, so that it only executes
+# the constchecker tests
+ CILIUM_CONSTCHECKER_BPF_OBJ=$BPFDIR/bpf_constchecker_self.o \
+	 go test github.com/cilium/cilium/pkg/constchecker -run '^Test$' -check.f 'ConstSuite.*'
```

Comment:
I _think_ there are Makefle environment variables set for `go` and `test`. I know this is a script but I'm wondering if it would make sense to keep them consistent.

Reply:
Indeed, they seem to be. My original thought was to add this test in the `Makefile` (defining the `CILIUM_CONSTCHECKER_BPF_OBJ` env variable would be enough), but this felt too intrusive/complex, so I decided to put it on a separate script. Since the const are fairly simple and have no other dependencies except for `cosntchecker`, I would think that running them as they are would be OK.